src/bitmap/ops.rs

```diff
@@ -155,6 +155,100 @@ impl RoaringBitmap {
     pub fn symmetric_difference_with(&mut self, other: &RoaringBitmap) {
         BitXorAssign::bitxor_assign(self, other)
     }
+
+    /// Computes the len of the intersection with the specified other bitmap without creating a
+    /// new bitmap.
+    ///
+    /// This is faster and more space efficient when you're only interested in the cardinality of
+    /// the intersection.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use roaring::RoaringBitmap;
+    ///
+    /// let rb1: RoaringBitmap = (1..4).collect();
+    /// let rb2: RoaringBitmap = (3..5).collect();
+    ///
+    ///
+    /// assert_eq!(rb1.intersection_len(&rb2), (rb1 & rb2).len());
+    /// ```
+    pub fn intersection_len(&self, other: &RoaringBitmap) -> u64 {
+        Pairs::new(&self.containers, &other.containers)
+            .map(|pair| match pair {
+                (Some(..), None) => 0,
+                (None, Some(..)) => 0,
+                (Some(lhs), Some(rhs)) => lhs.intersection_len(rhs),
+                (None, None) => unreachable!(),
```

Comment:
I like the suggestion since if we get this invalid state adding zero is idempotent, but this is still an invalid state. I think refactoring pairs to make this illegal state unrepresentable would be ideal. Out of scope to implement here of course, but curious to hear what you think.

```rust
enum Occurence<L, R> {
    Left(L),
    Right(R),
    Both(L, R)
}

Reply:
Yup, that's exactly [`EitherOrBoth` from itertools](https://docs.rs/itertools/latest/itertools/enum.EitherOrBoth.html) and I am interested in using this in the `Pairs` iterator 👍 .